src/Options/Generic.hs

```diff
@@ -218,31 +234,52 @@ auto = do
     language extension
 -}
 class ParseField a where
-    parseField
-        :: Maybe Text
+        --(forall f b. Options.Mod f b)
+    parseFieldH
+        :: HelpMod
```

Comment:
Instead of supplying a `(forall f b . Options.Mod f b)` here, could you instead supply a `Maybe Text` (where the `Text` is the help message and `Nothing` omits the message)?